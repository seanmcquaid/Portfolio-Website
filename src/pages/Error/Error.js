import React from 'react';
import styled from 'styled-components';
import { HeaderText, SubHeaderText } from 'components/components';
import { Link } from 'react-router-dom';

const Error = () => (
  <ErrorPageContainer>
    <HeaderText>Error</HeaderText>
    <SubHeaderText>
      There was a problem going to the page you typed in. Please go to the{' '}
      <HomePageLink to="/">Home Page</HomePageLink> and check out my portfolio!
    </SubHeaderText>
  </ErrorPageContainer>
);

const ErrorPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
`;

const HomePageLink = styled(Link)`
  text-decoration: none;
  color: #ffffffb6;
  &:hover {
    transition: 0.5s;
    color: #00000096;
    border-bottom: 2px solid #00000096;
  }
`;

export default Error;
