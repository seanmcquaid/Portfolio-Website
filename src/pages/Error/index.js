import React from 'react';
import styled, { keyframes } from 'styled-components';
import { HeaderText, SubHeaderText } from 'components';
import { Link } from 'react-router-dom';

const Error = () => (
  <ErrorPageContainer>
    <Header>
      <HeaderText>Error</HeaderText>
    </Header>
    <Main>
      <SubHeaderText>
        There was a problem going to the page you typed in. Please go to the{' '}
        <HomePageLink to="/" title="Home Page Link">
          Home Page
        </HomePageLink>{' '}
        and check out my portfolio!
      </SubHeaderText>
    </Main>
  </ErrorPageContainer>
);

const fadeInAnimation = keyframes`
    0% { 
        opacity: 0; 
    } 
    100% { 
        opacity: 1; 
    } 
`;

const ErrorPageContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  transition: 0.5s;
  animation: ${fadeInAnimation} ease 1s;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  @media (max-width: 975px) {
    justify-content: flex-start;
  }
`;

const Header = styled.header``;

const Main = styled.main``;

const HomePageLink = styled(Link)`
  text-decoration: none;
  color: #ffffffb6;
  &:hover {
    transition: 0.5s;
    color: #00000096;
    border-bottom: 2px solid #00000096;
  }
`;

export default Error;
