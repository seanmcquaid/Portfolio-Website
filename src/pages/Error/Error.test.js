import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import Error from './Error';

describe('<Error/>', () => {
  it('Matches snapshot', () => {
    const error = render(
      <Router>
        <Error />
      </Router>
    );
    expect(error).toMatchSnapshot();
  });
});
