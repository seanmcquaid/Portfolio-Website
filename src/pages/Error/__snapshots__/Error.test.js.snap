// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> Matches snapshot 1`] = `[Function]`;
